hathor/indexes/manager.py

```diff
@@ -87,29 +82,24 @@ def __init_checks__(self):
 
     def iter_all_indexes(self) -> Iterator[BaseIndex]:
         """ Iterate over all of the indexes abstracted by this manager, hiding their specific implementation details"""
-        for _, index in self._iter_all_indexes_with_filter():
-            yield index
-
-    def _iter_all_indexes_with_filter(self) -> Iterator[Tuple[_IndexFilter, BaseIndex]]:
-        """ Same as `iter_all_indexes()`, but includes a filter for what transactions an index is interested in."""
-        yield _IndexFilter.ALL, self.info
-        yield _IndexFilter.ALL, self.all_tips
-        yield _IndexFilter.ALL_BLOCKS, self.block_tips
-        yield _IndexFilter.VALID_TXS, self.tx_tips
-        yield _IndexFilter.ALL, self.sorted_all
-        yield _IndexFilter.ALL_BLOCKS, self.sorted_blocks
-        yield _IndexFilter.VALID_TXS, self.sorted_txs
-        yield _IndexFilter.ALL, self.height
+        yield self.info
+        yield self.all_tips
+        yield self.block_tips
+        yield self.tx_tips
+        yield self.sorted_all
+        yield self.sorted_blocks
+        yield self.sorted_txs
+        yield self.height
         if self.deps is not None:
-            yield _IndexFilter.ALL, self.deps
+            yield self.deps
         if self.mempool_tips is not None:
-            yield _IndexFilter.ALL, self.mempool_tips
+            yield self.mempool_tips
         if self.addresses is not None:
-            yield _IndexFilter.ALL, self.addresses
+            yield self.addresses
         if self.tokens is not None:
-            yield _IndexFilter.ALL, self.tokens
+            yield self.tokens
         if self.utxo is not None:
-            yield _IndexFilter.ALL, self.utxo
+            yield self.utxo
```

Comment:
Maybe could be simplified to something like:

```python
    def iter_all_indexes(self) -> Iterator[BaseIndex]:
        indexes = [
            self.info,
            self.all_tips,
            self.block_tips,
            self.tx_tips,
            self.sorted_all,
            self.sorted_blocks,
            self.sorted_txs,
            self.height,
            self.deps,
            self.mempool_tips,
            self.addresses,
            self.tokens,
            self.utxo
        ]
            
        return (index for index in indexes if index is not None)
```

Reply:
Yeah, that'd be better, I'll refactor it to `return filter(None, [self.info, self.all_tips, ...])`.

Reply:
Done.